Wide screen signaling (WSS) patch for vdr-dxr3 by Seppo Ingalsuo.
Apply to: vdr-dxr3 plugin sources
More info: http://thread.gmane.org/gmane.linux.vdr/17095
Note: you'll probably want to apply em8300-adv7170-wss.patch too
  in order to get support for this functionality in the driver.

Index: dxr3interface.c
===================================================================
RCS file: /cvsroot/dxr3plugin/dxr3/dxr3interface.c,v
retrieving revision 1.4.2.27
diff -u -r1.4.2.27 dxr3interface.c
--- dxr3interface.c	12 Dec 2005 17:16:15 -0000	1.4.2.27
+++ dxr3interface.c	12 Dec 2005 20:18:29 -0000
@@ -410,20 +410,39 @@
 void cDxr3Interface::SetAspectRatio(uint32_t ratio)
 {
     static int requestCounter = 0;
+    int wssmode;
+    int aspect;
 
     Lock();
 
     if (cDxr3ConfigData::Instance().GetForceLetterBox())
 	ratio = EM8300_ASPECTRATIO_16_9;
-    if (Setup.VideoFormat)
-	ratio = EM8300_ASPECTRATIO_4_3;
 
     if (!m_ExternalReleased && ratio != UNKNOWN_ASPECT_RATIO)
     {
 	if (ratio != m_aspectRatio && requestCounter > 50)
 	{
+	    if (Setup.VideoFormat)
+	    {
+		aspect = EM8300_ASPECTRATIO_4_3;
+#ifdef EM8300_IOCTL_SET_WSS
+		if (ratio == EM8300_ASPECTRATIO_16_9)
+		    wssmode = EM8300_WSS_16_9;
+		else
+		    wssmode = EM8300_WSS_OFF;
+		if (ioctl(m_fdControl, EM8300_IOCTL_SET_WSS, &wssmode) < 0)
+		{
+		    esyslog("dxr3: unable to set WSS: %m");
+		}
+#endif
+	    }
+	    else
+	    {
+		aspect = ratio;
+	    }
+
 	    requestCounter = 0;
-	    if (ioctl(m_fdControl, EM8300_IOCTL_SET_ASPECTRATIO, &ratio) < 0)
+	    if (ioctl(m_fdControl, EM8300_IOCTL_SET_ASPECTRATIO, &aspect) < 0)
 	    {
 		esyslog("dxr3: unable to set aspect ratio: %m");
 	    }
